projects/packages/identity-crisis/src/class-url-secret.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * IDC URL secret functionality.
+ *
+ * @package  automattic/jetpack-identity-crisis
+ */
+
+namespace Automattic\Jetpack\IdentityCrisis;
+
+use Jetpack_Options;
+use WP_Error;
+
+/**
+ * IDC URL secret functionality.
+ */
+class URL_Secret {
+
+	/**
+	 * The options key used to store the secret.
+	 */
+	const OPTION_KEY = 'identity_crisis_url_secret';
+
+	/**
+	 * Secret lifespan (5 minutes)
+	 */
+	const LIFESPAN = 300;
+
+	/**
+	 * The URL secret string.
+	 *
+	 * @var string|null
+	 */
+	private $secret = null;
+
+	/**
+	 * The URL secret expiration date in unix timestamp.
+	 *
+	 * @var string|null
+	 */
+	private $expires_at = null;
+
+	/**
+	 * Initialize the class.
+	 */
+	public function __construct() {
+		$secret_data = $this->fetch();
+
+		if ( $secret_data !== null ) {
+			$this->secret     = $secret_data['secret'];
+			$this->expires_at = $secret_data['expires_at'];
+		}
+	}
+
+	/**
+	 * Fetch the URL secret from the database.
+	 *
+	 * @return array|null
+	 */
+	private function fetch() {
+		$data = Jetpack_Options::get_option( static::OPTION_KEY );
```

Comment:
How come we don't store the secret as a transient with expiration 5mins?

Reply:
I opted against them because they are by definition unreliable: _["it is possible for the transient to not be available before the expiration time"](https://developer.wordpress.org/apis/transients/)_.
So I decided to use options instead, and manually invalidate them according to the expiration timestamp.